Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -604,23 +605,35 @@ void WiiSocket::DoSock(Request request, SSL_IOCTL type)
   pending_sockops.push_back(so);
 }
 
-void WiiSockMan::AddSocket(s32 fd)
+s32 WiiSockMan::AddSocket(s32 fd)
 {
   if (fd >= 0)
   {
-    WiiSocket& sock = WiiSockets[fd];
+    s32 wii_fd = 3;
+    while (WiiSockets.count(wii_fd) > 0)
+      wii_fd += 1;
+    WiiSocket& sock = WiiSockets[wii_fd];
     sock.SetFd(fd);
+    return wii_fd;
   }
+  return fd;
 }
 
 s32 WiiSockMan::NewSocket(s32 af, s32 type, s32 protocol)
 {
-  s32 fd = (s32)socket(af, type, protocol);
-  s32 ret = GetNetErrorCode(fd, "NewSocket", false);
-  AddSocket(ret);
+  s32 fd = static_cast<s32>(socket(af, type, protocol));
+  s32 wii_fd = AddSocket(fd);
+  s32 ret = GetNetErrorCode(wii_fd, "NewSocket", false);
   return ret;
 }
 
+s32 WiiSockMan::GetHostSocket(s32 wii_fd)
```

Comment:
this can be a const member function using `return WiiSockets.at(wii_fd).fd` in your if statement body.